src/storage/mvcc/txn.rs

```diff
@@ -125,27 +131,33 @@ impl<'a> MvccTxn<'a> {
                 };
             }
         }
+        self.writes.push(Modify::Put(CF_WRITE,
+                                     key.append_ts(self.start_ts),
+                                     Write::new(WriteType::Rollback, self.start_ts).to_bytes()));
         self.unlock_key(key.clone());
         Ok(())
     }
 
     pub fn gc(&mut self, key: &Key, safe_point: u64) -> Result<()> {
         let mut after_safe_point = false;
         let mut ts: u64 = u64::max_value();
-        while let Some((start, commit)) = try!(self.reader.seek_write(key, ts)) {
+        while let Some((commit, write)) = try!(self.reader.seek_write(key, ts)) {
             if !after_safe_point {
                 if commit <= safe_point {
                     // Set `after_safe_point` after the latest write after `safe_point`.
                     after_safe_point = true;
-                    // Latest write can be deleted if it's a `Modify::Delete`.
-                    if try!(self.reader.load_data(key, start)).is_none() {
-                        self.writes.push(Modify::Delete(CF_WRITE, key.append_ts(commit)));
+                    // Latest write can be deleted if its type is Delete/Rollback.
+                    match write.write_type {
+                        WriteType::Delete | WriteType::Rollback => {
```

Comment:
may still meet a slow prewrite after gc rollback. we should check in commit later.